### DISCUSSION TITLE
Erasure coding in CORTX

### DISCUSSION THREAD

**Discussion by mshaio** [Seagate/cortx#885] [Q&A] [answered]
Hi, 

I'm wondering if there is somewhere I can find out more details about the erasure coding being used in CORTX. For example:
1.  What's the upper limit of failures it's able to handle at any instant? 
2.  What erasure coding algorithm does CORTX use?

Thank you.

**Comment by mukul-seagate11** [comment]
@madhavemuri @huanghua78, can you address the queries asked?

**Comment by johnbent** [comment]
Hello @mshaio thanks for your question!  I don't know the answer to the upper limit but I know it can do N+K+S and N can be 1 or more and I know we are using K=2 and S=2.  Here is the code: https://github.com/Seagate/cortx-motr-galois

**Comment by mssawant** [comment, marked as answer]
@mshaio, Just extending John's answer,

>What's the upper limit of failures it's able to handle at any instant?

This depends on your configuration of the cluster. Motr uses [parity declustered layout](https://github.com/Seagate/cortx-motr/blob/main/layout/pdclust.h), also checkout Reed solomon algorithm. Fault tolerance is equivalent to the supported redundancy. In this case which is denoted by the value of `K` (no .of parity units). This can be extended to various failure domains based on your cluster's pool size.

>What erasure coding algorithm does CORTX use?

Presently CORTX-motr uses Parity declustered layout.
